### ISSUE TITLE
Feature request: Add configurable thresholds for the HTML output

### ISSUE THREAD

**Issue by rianquinn** [mozilla/grcov#631]
If there some way to make the following configurable?

```rust
    let conf = Config {
        hi_limit: 90.,
        med_limit: 75.,
        fn_hi_limit: 90.,
        fn_med_limit: 75.,
        branch_hi_limit: 90.,
        branch_med_limit: 75.,
        date: Utc::now(),
    };
```

The project that we use this for needs 100% for lines and doesn't care about the others.

**Comment by calixteman** [comment]
We could add an option `--config-file` for example which could take a json file (there's already a dep on serde-json) containing those options and some others we've in `--help`.
Would you mind making a PR for such thing ?

**Comment by rianquinn** [comment]
I have never programmed in Rust before, I am sure that I can manage, so yes, I will take a stab at it. The other feature that would be really nice is some way to see the branch coverage information. This is something that is needed for critical systems compliance, and there really isn't any open source tools that handle this properly. 

Specifically, with C++. Suppose you have the following:
```cpp
template<typename T>
bool foo(int val)
{
    if (val != 42) {
        return false;
    }

    return true;
}
```

This simple function appears to only have two branches (take or do not take the if), when in fact, GCOV will report two branches for every instantiated version of `foo`. For for example, `foo<int>()` is 2, and `foo<bool>()` would be another 2. GRCOV right now just shows the branch % with no real information about why the branches are lower than they should be. 

One feature that would really set GRCOV apart from all of the other tools would be useful information around branches. This would include:
- With all of the template versions merged, are all of the branch combinations hit? This is useful when you are using template to mock inputs. You don't care about testing every version of a branch, you just want to make sure that every branch is covered across all of your unit tests. 
- Information about which branches are not hit (given all possible combinations) and which specific branch was not taken. For example, LCOV and GCOVR will tell you branch 0, 1 and 3 were taken, but branch 2, 4 and 5 were not. The only way to figure out what this means is to look at the raw GCOV output and demangle the C++ functions names and then manually figure out which template instantiation belongs with each branch. Its not scalable at all. It would be great if GRCOV had the same branch taken, not taken, not executed data that LCOV and GCOVR have, but when you hover over one of the branch results, it highlights the specific portion of the branch that it belongs to, and then also shows a popup with the demangled C++ name of the function so that you can see specifically which version of template that branch is referring too. Without this information, branch data is simply impossible to use with C++. 

I don't have the ability to add that feature as that gets way too in the weeds of Rust and HTML, but man... that would be a pretty amazing feature. 

Also... what happened to the Windows releases? That used to be a pretty nice feature (meaning Windows in each release had pre-compiled binaries that you could just download and use).

**Comment by calixteman** [comment]
Too much stuff in your comment.
Please open an issue for each different requests: it's a way easier for us and I guess everybody.
About windows releases, we're aware of the breakages but we don't really have time to investigate that stuff, anyway we really should switch at some point to taskcluster but time is missing...

**Comment by marco-c** [comment]
Fixed by #783.
